MensaHub-Junction/src/main/java/de/olech2412/mensahub/junction/jpa/services/meals/MealsService.java

```diff
@@ -83,5 +83,21 @@ public void delete(Meal meal, Mensa mensa) {
         log.warn("Meal deleted: {} from {}", meal.getName(), mensa.getName());
     }
 
+    public List<Meal> findTop20DistinctMeals() {
+        return mealRepository.findTop20DistinctMeals();
+    }
+
+    public List<Meal> findTop20DistinctMealsForSubscribedMensa(Long userId) {
+        return mealRepository.findTop20DistinctMealsByUser(userId);
+    }
+
+    public List<String> findTop20DistinctMealNamesForSubscribedMensa(Long userId) {
+        return mealRepository.findTop20DistinctMealNamesByUser(userId);
+    }
+
+    public List<Meal> findTop20DistinctMealsExcludingGoudaForSubscribedMensa(Long userId) {
+        return mealRepository.findTop20DistinctMealsExcludingGoudaByUser(userId);
```

Comment:
Das ist für jetzt okay. Wir sollten diese Woche aber unbedingt mal die Top 20 Gerichte durchgucken und uns dann ein Regex bauen, was die meisten berücksichtigt. Es gibt auch Spirelli und Bandnudeln und sowas... Das ist bisschen scuffed über diese massive SQL Query. Sind ja auch nicht allzu viele Daten, also sollte ein simpler Stream easy reichen.

Reply:
Vllt. please noch in diesem PR fixen so mäßig, aber halt als stream und dann raussortieren: https://www.w3schools.com/java/java_regex.asp